### PR TITLE
Fix ModelYolo possible loss of data warning

### PR DESCRIPTION
**Pull request by Wovchena**
Cherry pick
6b126df45492c2316664163af6dcd921b2b58383

### CODE REVIEW

File: demos/common/cpp/models/include/models/detection_model_yolo.h [hunk 1]
```diff
@@ -27,14 +27,14 @@ class ModelYolo : public DetectionModel {
     class Region {
     public:
         int num = 0;
-        int classes = 0;
+        size_t classes = 0;
```
Comment (ivikhrev) [inline demos/common/cpp/models/include/models/detection_model_yolo.h:30]: It is still constructed from `int clasess` in 
https://github.com/openvinotoolkit/open_model_zoo/blob/b418da60e0c6aeb749b7acc9a9078b9c886a844b/demos/common/cpp/models/src/detection_model_yolo.cpp#L184-L188

File: demos/common/cpp/models/include/models/detection_model_yolo.h [hunk 2]
```diff
@@ -27,14 +27,14 @@ class ModelYolo : public DetectionModel {
...
         int coords = 0;
         std::vector<float> anchors;
-        int outputWidth = 0;
-        int outputHeight = 0;
+        size_t outputWidth = 0;
```
Comment (ivikhrev) [inline demos/common/cpp/models/include/models/detection_model_yolo.h:33]: It's too
https://github.com/openvinotoolkit/open_model_zoo/blob/b418da60e0c6aeb749b7acc9a9078b9c886a844b/demos/common/cpp/models/src/detection_model_yolo.cpp#L356-L357

File: demos/common/cpp/models/include/models/detection_model_yolo.h (hunk 1 above)
Comment (Wovchena) [reply demos/common/cpp/models/include/models/detection_model_yolo.h:30]: Updated

File: demos/common/cpp/models/include/models/detection_model_yolo.h (hunk 2 above)
Comment (Wovchena) [reply demos/common/cpp/models/include/models/detection_model_yolo.h:33]: Updated

**Review by ivikhrev — APPROVED** [review]

**Comment by vladimir-dudnik** [comment]
ignore non relevant issues

**Pull request merged**